test/role.test.js

```diff
@@ -102,35 +102,70 @@ describe('role model', function() {
-                if (err) return next(err);
-                assert.equal(users.length, 1);
-                assert.equal(users[0].id, user.id);
-                next();
-              });
-            },
-          ], done);
+          role.principals.create({principalType: User.modelName, principalId: user.id},
+          function(err, principal2) {
+            if (err) return done(err);
+            role.principals.create({principalType: 'incorrectPrincipalType', principalId: user.id},
+            function(err, principal3) {
+              if (err) return done(err);
+              async.parallel([
+                function(next) {
+                  Role.find(function(err, roles) {
+                    if (err) return next(err);
+                    assert.equal(roles.length, 1);
+                    assert.equal(roles[0].name, 'userRole');
+                    next();
+                  });
+                },
+                function(next) {
+                  role.principals(function(err, principals) {
+                    if (err) return next(err);
+                    assert.equal(principals.length, 3);
+                    assert.equal(principals[0].principalType, RoleMapping.USER);
+                    assert.equal(principals[0].principalId, user.id);
+                    assert.equal(principals[1].principalType, User.modelName);
+                    assert.equal(principals[1].principalId, user.id);
+                    assert.equal(principals[2].principalType, 'incorrectPrincipalType');
+                    assert.equal(principals[2].principalId, user.id);
+                    next();
+                  });
+                },
+                function(next) {
+                  role.users(function(err, users) {
+                    if (err) return next(err);
+                    assert.equal(users.length, 1);
+                    assert.equal(users[0].id, user.id);
+                    next();
+                  });
+                },
+                function(next) {
+                  principal.user(function(err, principalUser) {
+                    if (err) return next(err);
+                    assert(principalUser);
+                    assert.equal(principalUser.id, user.id);
+                    next();
+                  });
+                },
+                function(next) {
+                  principal2.user(function(err, principalUser) {
+                    if (err) return next(err);
+                    assert(principalUser);
+                    assert.equal(principalUser.id, user.id);
+                    next();
+                  });
+                },
+                function(next) {
+                  principal3.user(function(err, principalUser) {
+                    if (err) return next(err);
+                    assert(!principalUser);
+                    next();
+                  });
+                },
+              ], done);
+            });
+          });
         });
```

Comment:
1. creating 2 additional principals (RoleMapping instances) : one with a user model name as principalType, the other with an incorrect principalType
2. adding 3 unit-tests for principal.user() : the baseline unit-test with USER principalType was missing too